include/circt/Dialect/HW/HWAttributes.h

```diff
@@ -33,9 +33,9 @@ mlir::FailureOr<mlir::Type> evaluateParametricType(mlir::Location loc,
 
 /// Evaluates a parametric attribute (param.decl.ref/param.expr) based on a set
 /// of provided parameter values.
-mlir::FailureOr<mlir::APInt> evaluateParametricAttr(mlir::Location loc,
-                                                    mlir::ArrayAttr parameters,
-                                                    mlir::Attribute paramAttr);
+mlir::FailureOr<mlir::Attribute>
+evaluateParametricAttr(mlir::Location loc, mlir::ArrayAttr parameters,
+                       mlir::Attribute paramAttr);
```

Comment:
Attribute seems a bit broad here, but the paramAttr uses the same type. 🤷  Maybe this is something we should tighten up in the future.